packages/core/src/delegation.js

```diff
@@ -21,6 +21,71 @@ export const isLink =
  */
 export const isDelegation = proof => !Link.isLink(proof)
 
+/**
+ * Takes on or more delegations and returns all delegated capabilities in
+ * UCAN 0.10 format, expanding all the special forms like `with: ucan:*` and
+ * `can: *` to an explicit forms.
+ *
+ * @template {[API.Delegation, ...API.Delegation[]]} T
+ * @param {T} delegations
+ * @returns {API.InferAllowedFromDelegations<T>}
+ */
+export const allows = (...delegations) => {
+  /** @type {API.Allows} */
+  let allow = {}
+  for (const delegation of delegations) {
+    for (const capability of delegation.capabilities) {
+      // If uri is `ucan:*` then we include own capabilities along with
+      // delegated
+      const capabilities =
+        capability.with === 'ucan:*'
+          ? iterateCapabilities(delegation)
+          : [capability]
+
+      for (const { with: uri, can, nb } of capabilities) {
+        const resource = allow[uri] || (allow[uri] = {})
+        const abilities = resource[can] || (resource[can] = [])
+        abilities.push({ ...Object(nb) })
```

Comment:
Do we also have enough info here to include something like the relevant subset of `delegations` you'd need to exercise this capability? e.g. if so we could put a method on here that iterates them back out, in case that would be useful for putting only the proofs necessary on an invocation